chirho/explainable/internals/defaults.py

```diff
@@ -173,3 +174,90 @@ def _soft_neq_independent(support: constraints.independent, v1: T, v2: T, **kwar
     for _ in range(support.reinterpreted_batch_ndims):
         result = torch.sum(result, dim=-1)
     return result
+
+
+class InferSupports(pyro.poutine.messenger.Messenger):
```

Comment:
Can you move this to `handlers/components.py` so that it's available to user code and rename it to `ExtractSupports`? I know I suggested `InferSupports` in #448 but I think `ExtractSupports` is more accurate since the information from this handler is only available after it exits.

Reply:
done